File: 2264-largest-3-same-digit-number-in-string/2264-largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    bool isValid(string s){
        return (s[0] == s[1] && s[1] == s[2]);
    }
    
    string largestGoodInteger(string num) {
        
        string res = "";
        
        int n = num.size();
        
        for(int i = 0; i <= n - 3; i++){
            
            string s = num.substr(i, 3);
            
            if(isValid(s)){
                
                if(res.size() == 0)     res = s;
                
                else{
                    
                    int prev = stoi(res);
                    
                    int curr = stoi(s);
                    
                    if(prev < curr)     res = to_string(curr);
                    
                }
                
            }
            
        }
        
        return res;
        
    }
};

File: 2264-largest-3-same-digit-number-in-string/2264-largest-3-same-digit-number-in-string.java
class Solution {
    public String largestGoodInteger(String num) {
        
        int n = num.length();
        
        char maxChar = ' ';
        
        for(int i = 2; i < n; i++){
            
            if(num.charAt(i) == num.charAt(i - 1) && num.charAt(i - 1) == num.charAt(i - 2)){
                
                maxChar = (char)Math.max((char)maxChar, (char)num.charAt(i));
                
            }
            
        }
        
        if(maxChar == ' ')  return "";
        
        return new String(new char[]{maxChar, maxChar, maxChar});
    }
}